### PR TITLE
Fix Dismissible with CupertinoSlider as child not slidable

### PR DESCRIPTION
**Pull request by unhonin**
With the latest stable version of Flutter (3.0.4), the CupertinoSlider is not sliding properly when it has Dismissible as their parent, similar issues has been discussed in #105429, and has been fixed in 3.0.2, but only fixed the issue of Slider widget, while CupertinoSlider still has this problem.

This fix adds **gestureSettings** paramter to the initialization of **HorizontalDragGestureRecognizer**.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/slider.dart [hunk 1]
```diff
@@ -311,7 +312,8 @@ class _CupertinoSliderRenderObjectWidget extends LeafRenderObjectWidget {
       ..onChanged = onChanged
```
Comment (MitchellGoodwin) [inline packages/flutter/lib/src/cupertino/slider.dart:312]: There will need to be an `assert` for MediaQuery as well.

**Comment by MitchellGoodwin** [comment]
Thanks for the contribution. We'd love to have this in. This will need tests added in order to be landed.

**Comment by Hixie** [comment]
@unhonin Thanks for your contribution! Are you interested in writing tests for this as per the comments above?

**Comment by MitchellGoodwin** [comment]
@unhonin Thank you for the contribution. I'm going to close this PR to unblock the queue. If you'd ever like to pick this back up again, please feel free to open the PR.

**Pull request closed without merging**